creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -990,34 +994,31 @@ class SnapbackSM {
     }
 
     const startTimeMs = Date.now()
-    const endTimeMs = startTimeMs + MaxSyncMonitoringDurationInMs
+    const maxMonitoringTimeMs = startTimeMs + MaxSyncMonitoringDurationInMs
 
-    let additionalSyncRequired = true
-    let maxExportRangeExceeded = false
-    while (Date.now() < endTimeMs) {
+    let secondaryCaughtUp = false
+    let secondaryProgressed = false
+    let initialSecondaryClock = null
+    let finalSecondaryClock = null
+    while (Date.now() < maxMonitoringTimeMs) {
       try {
         const clockStatusResp = await axios(clockStatusRequestParams)
         const { clockValue: secondaryClockValue } = clockStatusResp.data.data
-
         this.log(`${logMsgString} secondaryClock ${secondaryClockValue}`)
 
+        // Record starting and current clock values for secondary to determine future action
+        if (initialSecondaryClock === null) {
```

Comment:
shouldn't this be if secondaryClockValue is null?

Reply:
because that's the axios value. right now this is checking if null === null

Reply:
no, this is just to record the secondary's initial clock value before monitoring, as a means of determining if it has progressed or not

Reply:
we wouldn't want to update this everytime secondaryClockVal is updated

Reply:
oh it's inside a while loop. of course. this is why while loops and i dont hang out much

Reply:
hahaha fair enough. i agree this function is not most readable - do you think it is worth refactoring / if so, how

Reply:
at minimum, i can document better if you think that'd help

Reply:
i honestly think my eye kind of skipped the while because of the back to back let declarations. maybe just add a newline before the while?

Reply:
this flow makes sense, just had to trace it. could we add comments on the variable declarations? I didnt realize that `secondaryProgressed = true` implies that a secondary had sync progress, but does not necessarily mean that it caught up

Reply:
i tried to make names more clear and add a bit more docs, hopefully it helps